src/backend/tray_icon/tray_icon.ts

```diff
@@ -0,0 +1,129 @@
+import { BrowserWindow, ipcMain, Menu, nativeImage, Tray } from 'electron'
+import i18next from 'i18next'
+import { RecentGame } from 'common/types'
+import { logInfo, LogPrefix } from '../logger/logger'
+import { handleProtocol } from '../protocol'
+import { getRecentGames } from '../recent_games/recent_games'
+import { handleExit, showAboutWindow } from '../utils'
+import { GlobalConfig } from '../config'
+import { iconDark, iconLight } from '../constants'
+import { backendEvents } from '../backend_events'
+
+export const initTrayIcon = async (mainWindow: BrowserWindow) => {
+  // create icon
+  const appIcon = new Tray(await getIcon(process.platform))
+
+  // helper function to set/update the context menu
+  const loadContextMenu = async (recentGames?: RecentGame[]) => {
+    if (!recentGames) {
+      recentGames = await getRecentGames({ limited: true })
+    }
+
+    appIcon.setContextMenu(contextMenu(mainWindow, recentGames))
+  }
+  loadContextMenu()
+
+  appIcon.setToolTip('Heroic')
+
+  // event listeners
+  appIcon.on('double-click', () => {
+    mainWindow.show()
+  })
+
+  ipcMain.on('changeLanguage', async () => {
+    loadContextMenu()
+  })
+
+  ipcMain.on('changeTrayColor', () => {
+    logInfo('Changing Tray icon Color...', { prefix: LogPrefix.Backend })
+    setTimeout(async () => {
+      appIcon.setImage(await getIcon(process.platform))
+      loadContextMenu()
+    }, 500)
+  })
+
+  backendEvents.on('recentGamesChanged', async (recentGames: RecentGame[]) => {
+    loadContextMenu(recentGames)
+  })
+
+  return appIcon
+}
+
+const iconSizesByPlatform = {
```

Comment:
Adding a type hint here might be a good idea
```suggestion
const iconSizesByPlatform: Record<'win32' | 'darwin' | 'linux', { width: number, height: number }> = {
```
(don't commit this directly, I doubt our linter would be happy with it)